### ISSUE TITLE
HTTP ERROR 500 /public/var/cache/prod cd

### ISSUE THREAD

**Issue by adamus007p** [PrestaShop/PrestaShop#37666]
### Prerequisites

- [x] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [x] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

I was asking to reopen issue ##30786 the HTTP ERROR 500 /public/var/cache/prod but no response 

I have noticed that from time to time in my server there is HTTP ERROR 500
It it related with /public/var/cache/prod that some files have no permission.

I have noticed that here some other users had the similar problems.

[image](https://user-images.githubusercontent.com/19956106/211434344-7c8207ae-b704-48ac-b40f-01ec8680ec47.png)

I can correct manually the permission to 0755 and the shop starts working OK.

namespaced_class_stub.php
appParameters.php
class_index.php
000 permissions ----

Why it may happened? Maybe it is some bug regards cache or Symfony?

On my end I have the same problem with Prestasop 1.7 and now with v8 with different php 7.3, 7.4, 8.0, 8.1.

Nginx Version Installed: 1.23.3 (in older versions too)
PHP Version Installed: 8.1.14
CentOS Version: CentOS 7 64bit
I use LEMP: centminmod but i saw that someone here had the similar problems with other enviroment.

The problem exist as from long time, so it is not introduced now.

The problem is exise from 1.7, 8,1 and now it exist in 8,2

Workaround for it is 
Only one working wordaround, not a fix is to set up:

opcache config to 0 (disable it)
opcache.revalidate_freq=0

save and then restart your php service.

but this is not a solution

### Steps to reproduce

It happen occiasionally when I doing something in BO. Eg. when I was adding new module autoupgrade and wanted to update PS to 8.0.1. 1.7.8.5 > 1.7.8.8 and in 1.7.8.8 > 8.0.1

the same in 8,2

### PrestaShop version(s) where the bug happened

8.2

### PHP version(s) where the bug happened

7.3, 7.4, 8.0, 8.1.

**Comment by kpodemski** [comment]
My understanding is that it only happens when you clear the cache?

**Comment by Quetzacoalt91** [comment]
(Removed the `autoupgrade` tag as it is a issue happening anywhere while running the shop)
It could be linked to your PHP configuration. What does this code return we you run it on your environment?

```php
<?php
var_dump(umask());
```

If it returns 0, this explains why you have files without any permissions.

**Comment by adamus007p** [comment]
@kpodemski yes it is when the cache is cleared. The problem occurs when the cache is cleared.

I believe that:
php see the file, but it is not exist in real, and clear cache do not link it or change or revalidate it and it make a crash


more over:
presta keep static value in orders it also go to cache and sometimes it see old value

**Comment by adamus007p** [comment]
Hello, I would like to add some logs, maybe it will be helpful [1prod-2024-12-20.log](https://github.com/user-attachments/files/18218721/1prod-2024-12-20.log)


@kpodemski I can say it can happen when you will change something in BO, disable module, update it. Not only when you will clear the cache. Yesterday I had a such situaction.

**Comment by jromero-onclick** [comment]
Same Issue. It happens when we change some key translation from translations page

**Comment by dbouariu75-wq** [comment]
Same issue. On translation it crashes. Error 500.
It happens on every translation I make.

**Comment by boo-code** [comment]
Found the mechanism, and the umask hypothesis on this thread is inverted: `umask()` returning 0 is the
one value that is safe. The files end up mode 000 only when the umask masks out **all** of `0666`.

`Symfony\Component\HttpKernel\Kernel::dumpContainer()` chmods every file it dumps into
`var/cache/<env>/`:

```php
foreach ($content as $file => $code) {
    $fs->dumpFile($dir.$file, $code);
    @chmod($dir.$file, 0666 & ~umask());
}
```

Measured that expression on PHP 8.1 across the umasks a host can plausibly set:

```
umask 0000 -> 0666   readable
umask 0002 -> 0664   readable
umask 0022 -> 0644   readable
umask 0027 -> 0640   readable
umask 0077 -> 0600   readable
umask 0177 -> 0600   readable
umask 0666 -> 0000   NOT readable
umask 0777 -> 0000   NOT readable
```

So the condition is `umask & 0666 === 0666`. The `@` suppresses nothing here - the chmod succeeds, it
just sets no permission bits - which is why there is no error in the log, only a 500 on the next
request that tries to include the file.

That matches every symptom reported: it happens after the container is re-dumped, which is what
clearing the cache, saving a translation, toggling a module and several back-office settings all do.
It is not specific to one file; `appParameters.php` and `class_index.php` are written through
`Filesystem::dumpFile()`, which under such a umask throws a visible `IOException` instead of leaving an
unreadable file, so those two fail loudly rather than silently.

To confirm it on the affected server, from the same PHP-FPM pool that serves the shop, not from CLI:

```php
<?php printf("%04o\n", umask());
```

Anything whose `& 0666` is `0666` reproduces it. The fix belongs in the pool configuration
(`php_admin_value` / the process manager's umask) or in the front controller, and PrestaShop already
ships the hint for the CLI half of the problem - `bin/console:14` carries a commented
`// umask(0000);` for exactly this.

Core deliberately sets no umask of its own, and that should stay: forcing `umask(0000)` would make the
cache of every correctly configured install world-writable to fix a misconfigured one.
